web3_proxy/src/frontend/users/referral.rs

```diff
@@ -70,3 +75,129 @@ pub async fn user_referral_link_get(
     let response = (status_code, Json(response_json)).into_response();
     Ok(response)
 }
+
+#[debug_handler]
+pub async fn user_used_referral_stats(
+    Extension(app): Extension<Arc<Web3ProxyApp>>,
+    TypedHeader(Authorization(bearer)): TypedHeader<Authorization<Bearer>>,
+    Query(_params): Query<HashMap<String, String>>,
+) -> Web3ProxyResponse {
+    // First get the bearer token and check if the user is logged in
+    let (user, _semaphore) = app.bearer_is_authorized(bearer).await?;
+
+    let db_replica = app
+        .db_replica()
+        .context("getting replica db for user's revert logs")?;
+
+    // Get all referral records associated with this user
+    let referrals = referee::Entity::find()
+        .filter(referee::Column::UserId.eq(user.id))
+        .find_also_related(referrer::Entity)
+        .all(db_replica.conn())
+        .await?;
+
+    // For each related referral person, find the corresponding user-address
+    #[derive(Debug, Serialize)]
+    struct Info {
+        credits_applied_for_referee: bool,
+        credits_applied_for_referrer: Decimal, // Probably make this a float instead (approximate anyways)
+        referral_start_date: DateTime,
+        used_referral_code: String,
+        referrer_address: String, // Address of the referrer
```

Comment:
I'm wondering if we actually want to expose the address. Just because I refer you doesn't mean I want you to be able to track me on chain.

Reply:
I had the same thought, I will remove this for now then